Core/src/Geometry/Blueprint.cpp

```diff
@@ -231,63 +293,22 @@ std::unique_ptr<TrackingGeometry> Blueprint::construct(
 
   ACTS_DEBUG(prefix() << "Assigning volume IDs for remaining volumes");
 
-  TrackingVolume *currentVolume = nullptr;
-  GeometryIdentifier::Value iportal = 0;
-  GeometryIdentifier::Value isensitive = 0;
-  world->apply(overloaded{
-      [&](TrackingVolume &volume) {
-        iportal = 0;
-        isensitive = 0;
-        currentVolume = &volume;
-
-        if (volume.geometryId() != GeometryIdentifier{}) {
-          return;
-        }
-
-        auto it = std::ranges::find(volumesById, nullptr);
-        if (it == volumesById.end()) {
-          ACTS_ERROR(prefix() << "No free volume IDs left, all "
-                              << volumesById.size() << " are used");
-          // @TODO: Maybe link to documentation about this
-          throw std::logic_error("No free volume IDs left");
-        }
-
-        auto id = GeometryIdentifier().withVolume(
-            std::distance(volumesById.begin(), it) + 1);
-
-        ACTS_DEBUG(prefix() << "Assigning volume ID " << id << " for "
-                            << volume.volumeName());
-        volume.assignGeometryId(id);
-
-        *it = &volume;
-      },
-      [&](::Acts::Portal &portal) {
-        if (currentVolume == nullptr) {
-          // This should not really happen
-          ACTS_ERROR(prefix() << "No current volume found");
-          throw std::logic_error("No current volume found");
-        }
-
-        iportal += 1;
-        auto id = currentVolume->geometryId().withBoundary(iportal);
-        ACTS_VERBOSE(prefix() << "Assigning portal ID: " << id);
-        portal.surface().assignGeometryId(id);
-      },
-      [&](Surface &surface) {
-        if (currentVolume == nullptr) {
-          ACTS_ERROR(prefix() << "No current volume found");
-          throw std::logic_error("No current volume found");
-        }
-
-        if (surface.geometryId() != GeometryIdentifier{}) {
-          return;
-        }
-
-        isensitive += 1;
-        auto id = currentVolume->geometryId().withSensitive(isensitive);
-        ACTS_VERBOSE(prefix() << "Assigning surface ID: " << id);
-        surface.assignGeometryId(id);
-      }});
+  std::array<GeometryIdentifier, GeometryIdentifier::getMaxVolume()>
+      usedGeoIds{};
+  usedGeoIds.fill(GeometryIdentifier{});
+
+  std::ranges::for_each(volumesById,
+                        [&usedGeoIds](const TrackingVolume *volume) {
```

Comment:
Nitpick: in this case, I think a regular for loop is more readable than `for_each`.